### ISSUE TITLE
Animations should stop until the next turn of the event loop if script reads layout properties

### ISSUE THREAD

**Issue by pcwalton** [servo/servo#5533]
roc pointed out a potential web-compat problem in off-main-thread animations. This could fail in Servo, whereas it succeeds in all other browser engines:

    let el = ... some element ... ;
    let r0 = el.getBoundingClientRect();
    let r1 = el.getBoundingClientRect();
    assert(r0.left == r1.left);

This is because the animation refresh could tick in between `let r0 = ...` and `let r1 = ...` and mess with the layout.

The easiest way to fix this is to stop all animations until the next turn of the event loop if script performs layout queries.

**Comment by nox** [comment]
@pcwalton Wasn't this fixed?

**Comment by jdm** [comment]
I do not believe we have any code that stops animations from progressing if script performs a layout query.

**Comment by mrobinson** [comment]
We no longer have OTM animations.
